### PR TITLE
Automatically analyze in auto-regressive setting

### PR DESCRIPTION
**Pull request by mgoin**
Take the implementation for benchmark in https://github.com/neuralmagic/deepsparse/pull/1142 and apply to debug_analysis. Also update help usage and change default seq_len to 512

### CODE REVIEW

File: src/deepsparse/benchmark/benchmark_model.py [hunk 1]
```diff
@@ -153,25 +167,21 @@ def parse_args():
...
         "-seq_len",
         "--sequence_length",
         type=int,
-        default=2048,
-        help="The sequence length to run the "
-        "KV cache supported model benchmarks for. "
-        "Must be greater than 0, default is 2048",
+        default=512,
+        help="The sequence length to run the KV cache supported model "
+        "benchmarks for. Must be 1 <= seq_len, default is 512",
```
Comment (tlrmchlsmth) [inline src/deepsparse/benchmark/benchmark_model.py:176]: >Must be 1 <= seq_len,

do normal people write it this way?

File: src/deepsparse/debug_analysis.py [hunk 2]
```diff
@@ -49,10 +54,25 @@
...
                         Override the shapes of the inputs, i.e. -shapes
                         "[1,2,3],[4,5,6],[7,8,9]" results in input0=[1,2,3]
                         input1=[4,5,6] input2=[7,8,9]
+  --use-kvcache         Enable internal KVCache
+  --kv-cache-prev-num-tokens KV_CACHE_PREV_NUM_TOKENS
+                        KVCache: The amount of previous tokens that will be
+                        read from the external KV cache on the first inference
+  --kv-cache-num-frozen-tokens KV_CACHE_NUM_FROZEN_TOKENS
+                        KVCache: The amount of first tokens that we want to
+                        keep permanently in the KV cache
```
Comment (tlrmchlsmth) [inline src/deepsparse/debug_analysis.py:75]: Could you default these to 0? I don't see us setting them very often.

File: src/deepsparse/debug_analysis.py [hunk 3]
```diff
@@ -307,10 +349,31 @@ def main():
...
+    print(f"Analyzing model: {orig_model_path}")
 
     batch_size = args.batch_size
 
+    if has_model_kv_cache(model_path):
+        if batch_size != 1:
+            raise ValueError(
+                "Unable to run models with KV cache support "
+                "for batch size different than one."
+                "Please set batch size to 1 and try again"
```
Comment (tlrmchlsmth) [inline src/deepsparse/debug_analysis.py:361]: Should we use `--use-kvcache` here instead of querying `has_model_kv_cache`?

Comment (mgoin) [reply src/deepsparse/debug_analysis.py:361]: `--use-kvcache` just enables the internal kv cache and we want to be able to measure the difference. Probably `--use-kvcache` should be renamed

File: src/deepsparse/debug_analysis.py (hunk 2 above)
Comment (mgoin) [reply src/deepsparse/debug_analysis.py:75]: Yup good call and cleans up the logic

File: src/deepsparse/benchmark/benchmark_model.py (hunk 1 above)
Comment (mgoin) [reply src/deepsparse/benchmark/benchmark_model.py:176]: lol i was making it the same as input_ids_length

**Review by tlrmchlsmth — APPROVED** [review]
lgtm, thanks!

**Pull request merged**